File: internal/db/pagination.go
package db

// PaginationParams holds limit and offset for paginated queries.
type PaginationParams struct {
	Limit  int64 `json:"limit"`
	Offset int64 `json:"offset"`
}

// PaginatedResponse wraps a page of results with pagination metadata.
type PaginatedResponse[T any] struct {
	Data   []T   `json:"data"`
	Total  int64 `json:"total"`
	Limit  int64 `json:"limit"`
	Offset int64 `json:"offset"`
}

File: sdks/swift/Sources/Modula/Pagination.swift
public struct PaginationParams: Sendable {
    public let limit: Int64
    public let offset: Int64

    public init(limit: Int64, offset: Int64) {
        self.limit = limit
        self.offset = offset
    }
}

public struct PaginatedResponse<T: Decodable & Sendable>: Decodable, Sendable {
    public let data: [T]
    public let total: Int64
    public let limit: Int64
    public let offset: Int64

    enum CodingKeys: String, CodingKey {
        case data
        case total
        case limit
        case offset
    }
}
